cpp11/Ice/interceptor/Client.cpp

```diff
@@ -0,0 +1,181 @@
+//
+// Copyright (c) ZeroC, Inc. All rights reserved.
+//
+
+#include <Ice/Ice.h>
+#include <Interceptor.h>
+
+using namespace std;
+
+int run(const shared_ptr<Ice::Communicator>&);
+
+int
+main(int argc, char* argv[])
+{
+    int status = 0;
+
+    try
+    {
+        //
+        // CommunicatorHolder's ctor initializes an Ice communicator,
+        // and its dtor destroys this communicator.
+        //
+        Ice::CommunicatorHolder ich(argc, argv, "config.client");
+        auto communicator = ich.communicator();
+
+        //
+        // The communicator intialization removes all Ice-related arguments from argc/argv.
+        //
+        if(argc > 1)
+        {
+            cerr << argv[0] << ": too many arguments" << endl;
+            status = 1;
+        }
+        else
+        {
+            status = run(communicator);
+        }
+    }
+    catch(const exception& ex)
+    {
+        cerr << ex.what() << endl;
+        status = 1;
+    }
+
+    return status;
+}
+
+void menu();
+
+int
+run(const shared_ptr<Ice::Communicator>& communicator)
+{
+    auto context = communicator->getImplicitContext();
```

Comment:
It's better to not define variables too early as it's not clear what this variable is about and when it will be used... best if to define it below when it's needed in each of `if` block... it's actually only needed for `release-token`.

Reply:
`get-token` doesn't really need a convenience context variable, it can just use `communicator->getImplicitContext()->put("accessToken", token);`

`getImplicitContext` is cheap and IMO it's best to privilege clarity demos. Defining `context` where it's actually used is better than defining it somewhere where it's no longer visible at the point of use.